new-log-viewer/src/services/decoders/JsonlDecoder/index.ts

```diff
@@ -0,0 +1,244 @@
+import {Dayjs} from "dayjs";
+
+import {Nullable} from "../../../typings/common";
+import {
+    Decoder,
+    DecodeResultType,
+    FilteredLogEventMap,
+    JsonlDecoderOptionsType,
+    LogEventCount,
+} from "../../../typings/decoders";
+import {Formatter} from "../../../typings/formatters";
+import {JsonValue} from "../../../typings/js";
+import {
+    INVALID_TIMESTAMP_VALUE,
+    JsonLogEvent,
+    LOG_LEVEL,
+    LogLevelFilter,
+} from "../../../typings/logs";
+import LogbackFormatter from "../../formatters/LogbackFormatter";
+import {
+    convertToDayjsTimestamp,
+    convertToLogLevelValue,
+    isJsonObject,
+} from "./utils";
+
+
+/**
+ * A decoder for JSONL (JSON lines) files that contain log events. See `JsonlDecoderOptionsType` for
+ * properties that are specific to log events (compared to generic JSON records).
+ */
+class JsonlDecoder implements Decoder {
+    static #textDecoder = new TextDecoder();
+
+    #dataArray: Nullable<Uint8Array>;
+
+    #logLevelKey: string;
+
+    #timestampKey: string;
+
+    #logEvents: JsonLogEvent[] = [];
+
+    #filteredLogEventMap: FilteredLogEventMap = null;
+
+    #invalidLogEventIdxToRawLine: Map<number, string> = new Map();
+
+    #formatter: Formatter;
+
+    /**
+     * @param dataArray
+     * @param decoderOptions
+     */
+    constructor (dataArray: Uint8Array, decoderOptions: JsonlDecoderOptionsType) {
+        this.#dataArray = dataArray;
+        this.#logLevelKey = decoderOptions.logLevelKey;
+        this.#timestampKey = decoderOptions.timestampKey;
+        this.#formatter = new LogbackFormatter({formatString: decoderOptions.formatString});
+    }
+
+    getEstimatedNumEvents (): number {
+        return this.#logEvents.length;
+    }
+
+    getFilteredLogEventMap (): FilteredLogEventMap {
+        return this.#filteredLogEventMap;
+    }
+
+    setLogLevelFilter (logLevelFilter: LogLevelFilter): boolean {
+        this.#filterLogEvents(logLevelFilter);
+
+        return true;
+    }
+
+    build (): LogEventCount {
+        this.#deserialize();
+
+        const numInvalidEvents = Array.from(this.#invalidLogEventIdxToRawLine.keys()).length;
```

Comment:
```suggestion
        const numInvalidEvents = this.#invalidLogEventIdxToRawLine.size;
```
I agree with the Rabbit on this one.